pkg/codegen/pcl/binder_test.go

```diff
@@ -62,8 +63,22 @@ func TestBindProgram(t *testing.T) {
 					t.Fatalf("failed to parse files: %v", parser.Diagnostics)
 				}
 
-				_, diags, err := BindProgram(parser.Files, PluginHost(utils.NewHost(testdataPath, fileToMockPlugins[fileName])))
-				assert.NoError(t, err)
+				var bindError error
+				var diags hcl.Diagnostics
+				if fileName == "simple-range.pp" {
+					// simple-range.pp requires AllowMissingVariables
```

Comment:
Why? I don't see a missing var?

Reply:
It fails when the `range` expression is constant and we then use it in `range.value`. This is the error:
> === FAIL: codegen/pcl TestBindProgram/simple-range.pp (0.85s)
    binder_test.go:68: failed to bind program: simple-range.pp:5,28-33: undefined variable range; 
    --- FAIL: TestBindProgram/simple-range.pp (0.85s)

I need to look into this at some point because this is what tf2pulumi generates from terraform example:
Given
```terraform
resource "aws_rds_cluster" "default" {
  cluster_identifier = "aurora-cluster-demo"
  availability_zones = ["us-west-2a", "us-west-2b", "us-west-2c"]
  database_name      = "mydb"
  master_username    = "foo"
  master_password    = "barbut8chars"
}

resource "aws_rds_cluster_instance" "cluster_instances" {
  count              = 2
  identifier         = "aurora-cluster-demo-${count.index}"
  cluster_identifier = aws_rds_cluster.default.id
  instance_class     = "db.r4.large"
  engine             = aws_rds_cluster.default.engine
  engine_version     = aws_rds_cluster.default.engine_version
}
```
tf2pulumi generates PCL
```
resource default "aws:rds/cluster:Cluster" {
  clusterIdentifier = "aurora-cluster-demo"
  availabilityZones = ["us-west-2a", "us-west-2b", "us-west-2c"]
  databaseName      = "mydb"
  masterUsername    = "foo"
  masterPassword    = "barbut8chars"
}
resource clusterInstances "aws:rds/clusterInstance:ClusterInstance" {
options {
  range              = 2

}
  identifier         = "aurora-cluster-demo-${range.value}"
  clusterIdentifier = default.id
  instanceClass     = "db.r4.large"
  engine             = default.engine
  engineVersion     = default.engineVersion
}
```

Reply:
That looks like the binder doesn't understand options.range correctly, can you open an issue for that and put a "TODO issue/XXX" on this bit of binder_test.go

Reply:
Made an issue and added a TODO comment ✅